src/NuGet.Core/NuGet.ProjectManagement/NuGetProject.cs

```diff
@@ -35,6 +37,14 @@ protected NuGetProject(Dictionary<string, object> metadata)
             get { return InternalMetadata; }
         }
 
+        public string Guid
```

Comment:
I'm not disagreeing with your feedback, just explaining the use case.

The id should represent the batch not the project, there are other indicators to what project this runs on